src/include/planner/operator/simple/logical_simple.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include "planner/operator/logical_operator.h"
+
+namespace kuzu {
+namespace planner {
+
+class LogicalSimple : public LogicalOperator {
+public:
+    LogicalSimple(LogicalOperatorType operatorType,
+        std::shared_ptr<binder::Expression> outputExpression)
+        : LogicalOperator{operatorType}, outputExpression{std::move(outputExpression)} {}
+    LogicalSimple(LogicalOperatorType operatorType,
+        std::vector<std::shared_ptr<LogicalOperator>> plans,
+        std::shared_ptr<binder::Expression> outputExpression)
+        : LogicalOperator{operatorType, std::move(plans)},
+          outputExpression{std::move(outputExpression)} {}
+
+    void computeFactorizedSchema() override {
```

Comment:
move to cpp. Also you seem to flip the logic in `computeFactorizedSchema` and `computeFlatSchema`.

In `computeFlatSchema`, you should insert to group 0.
In `computeFactorizedSchema`, you should insert to a new group pos.